pkg/config/generate.go

```diff
@@ -227,13 +229,18 @@ func Generate(cs *api.OpenShiftManagedCluster, pluginConfig api.PluginConfig) (e
 		if cert.signingKey == nil && cert.signingCert == nil {
 			cert.signingKey, cert.signingCert = c.Certificates.Ca.Key, c.Certificates.Ca.Cert
 		}
-		if *cert.key != nil && *cert.cert != nil &&
-			((*cert.cert).CheckSignatureFrom(cert.signingCert) == nil || cert.selfSign) {
-			continue
-		}
-		if *cert.key, *cert.cert, err = tls.NewCert(cert.cn, cert.organization, cert.dnsNames, cert.ipAddresses, cert.extKeyUsage, cert.signingKey, cert.signingCert, cert.selfSign); err != nil {
+		var k *rsa.PrivateKey
+		var c *x509.Certificate
+		if k, c, err = tls.NewCert(cert.cn, cert.organization, cert.dnsNames, cert.ipAddresses, cert.extKeyUsage, cert.signingKey, cert.signingCert, false); err != nil {
 			return
 		}
+		// compare certificate and replace if update is needed
+		// if at any point we start using self-sign certificates again
+		// (see console certificate comment), logic inside certEqual needs
+		// to be updated for this.
+		if !certEqual(c, *cert.cert) {
```

Comment:
This is the sole cpu-intensive part of the plugin.  It would be far better CPU wise to look at the old certificate and see if it is inconsistent with the new inputs, and only then generate a new certificate if needed.

Reply:
the challenge is that if lets say I change API FQDN. I have `config` object with Old values. And I have 18 certificates, without any distinguished mapping which certificate is using FQDN in any field. 
So when logic comes to `Router` certificate - It will not match any of the values, so logic would say that it needs to be regenerated. When in reality - it does not.

If we want to compare certificates with values without generation - certificates should have list of attributes assigned to them to know which certs should contain one or another attribute. 

I might be getting this from the wrong angle, but each time I dont generate certificate I get "mapping problem" to check right values with right certificates.